reference_designs/ipu-xrt/vision_pipelines/edge_detect/aie2_edgeDetect.py

```diff
@@ -0,0 +1,174 @@
+#
+# This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# (c) Copyright 2021 Xilinx Inc.
+
+import sys
+
+from aie.ir import *
+from aie.dialects.func import *
+from aie.dialects.scf import *
+from aie.dialects.aie import *
+from aie.dialects.aiex import *
```

Comment:
Since this is a teaching exercise, it might be better to avoid these wild-card import to makes things clean.
Imagine a casual reader not knowing about AIE(X), MLIR, compilation... :-)
At least you could use some namespace shortcut if you dislike spelling `aie.dialects.aie.` and others all over the place.
For some context, https://peps.python.org/pep-0008
> Wildcard imports (`from <module> import *`) should be avoided, as they make it unclear which names are present in the namespace, confusing both readers and many automated tools. There is one defensible use case for a wildcard import, which is to republish an internal interface as part of a public API (for example, overwriting a pure Python implementation of an interface with the definitions from an optional accelerator module and exactly which definitions will be overwritten isn’t known in advance).
> 
> When republishing names this way, the guidelines below regarding public and internal interfaces still apply.